### DISCUSSION TITLE
Bitcoinize/stabilize

### DISCUSSION THREAD

**Discussion by jooray** [get10101/10101#1307] [General]
There is a bitcoinize/stabilize button. Can you alter the position in stable tab? Say I stabilize 10$. Is there a way to change that to 20$ (opening another short for 10$)? I only see the bitcoinize so it seems I need to exit the whole position.

You can easily enter and exit the stable position by opening another position. When you are renewing, you can consolidate all to one position, but other than that, you should be able to easily extend or reduce the stabilized position.

Also, it would be great to be able to pay a lightning invoice from the stable balance (that would be opening a long - bitcoinize - in the same amount as the invoice). And also when receiving payments over lightning, I should be able to specify that I want to automatically stabilize it, so when the invoice is paid, it is converted to synthetic stable.

Kollider Wallet does it similarly and I think it is a really great user experience. You just see USD balance and sats balance without needing to confuse users with "bitcoinize/stabilize". You can even call it "exchange" - I want to exchange BTC to USD. Although that might have some legal implications. But it is really great user experience...

**Comment by Restioson** [comment]
I don't believe this will be possible until we can extend positions, but I do believe it's planned

**Reply by jooray** [reply]
Opening a new one should be enough.

looking forward!

**Comment by holzeis** [comment]
> There is a bitcoinize/stabilize button. Can you alter the position in stable tab? Say I stabilize 10$. Is there a way to change that to 20$ (opening another short for 10$)? I only see the bitcoinize so it seems I need to exit the whole position.

Yes at the moment you will have to close and open the position again, since we do not support resizing nor multiple positions at the moment.

> You can easily enter and exit the stable position by opening another position. When you are renewing, you can consolidate all to one position, but other than that, you should be able to easily extend or reduce the stabilized position.

I think we need to differentiate here between the logical representation of a position and how it is mapped to a dlc. We could show multiple positions reflected in a single DLC. However, that may become a little bit more complex to manage.

> Also, it would be great to be able to pay a lightning invoice from the stable balance (that would be opening a long - bitcoinize - in the same amount as the invoice). And also when receiving payments over lightning, I should be able to specify that I want to automatically stabilize it, so when the invoice is paid, it is converted to synthetic stable.

Yes that's on our roadmap. Without having to manage multiple DLCs that would be made possible with resizing an existing DLC.

> Kollider Wallet does it similarly and I think it is a really great user experience. You just see USD balance and sats balance without needing to confuse users with "bitcoinize/stabilize". You can even call it "exchange" - I want to exchange BTC to USD. Although that might have some legal implications. But it is really great user experience...

I like the term "exchange". Although there isn't really an exchange happening. Maybe "transform"?

**Reply by jooray** [reply]
> I think we need to differentiate here between the logical representation of a position and how it is mapped to a dlc. We could show multiple positions reflected in a single DLC. However, that may become a little bit more complex to manage.

Nope, I would do the exact opposite - always show a single position (they add up to one position, it does not matter much how they are represented in DLCs), but have multiple DLCs backing them. If I stabilize $50 and want to add $50, it's easiest to open another short DLC for $50 and then merge them at renewal time.

> I like the term "exchange". Although there isn't really an exchange happening. Maybe "transform"?

Yes, transform is a perfect name. I think "bitcoinize"/"stabilize" is too difficult for people, they want to transform their BTC to USD (although in the background, it's still BTC with a matching short position).

**Reply by holzeis** [reply]
> Nope, I would do the exact opposite - always show a single position (they add up to one position, it does not matter much how they are represented in DLCs), but have multiple DLCs backing them. If I stabilize $50 and want to add $50, it's easiest to open another short DLC for $50 and then merge them at renewal time.

I think we are on the same page then. Granted that could be implemented with multiple DLCs, however implementing it by adapting an existing DLC seems to me far more efficient.
